common/src/main/java/io/pravega/common/concurrent/ExecutorServiceFactory.java

```diff
@@ -246,13 +237,13 @@ private void logNewThreadPoolCreated(String poolName) {
     }
 
     @VisibleForTesting
-    void checkThreadPoolLeak(ThreadPoolExecutor e, Exception stackTraceEx) {
+    void checkThreadPoolLeak(ThreadPoolScheduledExecutorService e, Exception stackTraceEx) {
         if (this.detectionLevel == ThreadLeakDetectionLevel.None) {
             // Not doing anything in this case.
             return;
         }
 
-        if (!e.isShutdown() || !e.isTerminated()) {
+        if (!e.isShutdown()) {
```

Comment:
Both are passed to the impl threadpool. But they have different meaning. Shutdown means shutdown was called. Terminated means all the treads have exited their run methods and stopped. It is possible for the running threads to not hold a reference to the pool of which they are a member. As such it may be the case that the pool is GCed after being properly closed before the thread has actually exited. This was causing some spurious test failures.